### PR TITLE
Initial imports support (for proto & graphql plugins).

### PR DESCRIPTION
**Pull request by damour**

### CODE REVIEW

File: cmd/go2gql/main.go [hunk 1]
```diff
@@ -36,6 +36,12 @@ func main() {
 			if err != nil {
 				panic(err)
 			}
+
+			err = gc.ParseImports()
+			if err != nil {
+				panic(err)
```
Comment (saturn4er) [inline cmd/go2gql/main.go]: panic?

File: generator/config.go [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+	for _, importPath := range gc.Imports {
+		normalizedPath := os.ExpandEnv(importPath)
+		normalizedPath, err := filepath.Abs(normalizedPath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to make normalized path '%s' absolute", normalizedPath)
+		}
+
+		cfgFile, err := os.Open(normalizedPath)
+		if err != nil {
+			return err
```
Comment (saturn4er) [inline generator/config.go]: need to wrap error
```suggestion
			return errors.Wrapf(err, "failed to open import '%s' file", normalizedPath)
```

File: cmd/go2gql/main.go [hunk 3]
```diff
@@ -36,6 +36,12 @@ func main() {
 			if err != nil {
 				panic(err)
 			}
+
+			err = gc.ParseImports()
+			if err != nil {
```
Comment (incu6us) [inline cmd/go2gql/main.go]: probably, inline declaration should be better?

File: generator/config.go [hunk 4]
```diff
@@ -1,8 +1,48 @@
...
+		}
+		cfg, err := ioutil.ReadAll(cfgFile)
+		if err != nil {
+			return err
```
Comment (saturn4er) [inline generator/config.go]: you can use `ioutil.ReadFile` instead of open and read it separately

File: generator/plugins/graphql/plugin.go [hunk 5]
```diff
@@ -35,6 +35,71 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
 	}
 	p.schemaConfigs = cfgs
 	p.generateCfg = config
+
+	err = p.parseImports()
+	if err != nil {
```
Comment (incu6us) [inline generator/plugins/graphql/plugin.go]: inline?

File: generator/plugins/proto2gql/plugin.go [hunk 6]
```diff
@@ -39,12 +39,35 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
...
 		return errors.Wrap(err, "failed to normalize config paths")
 	}
+
+	return nil
+}
+
+func (p *Plugin) parseImports() error {
+	for _, pluginsConfigsImports := range p.generateConfig.PluginsConfigsImports {
+		cfg := new([]*ProtoFileConfig)
+		err := mapstructure.Decode(pluginsConfigsImports["proto2gql_files"], cfg)
```
Comment (saturn4er) [inline generator/plugins/proto2gql/plugin.go]: can you make a constant for "proto2gql_files" string?

File: generator/plugins/graphql/plugin.go [hunk 7]
```diff
@@ -35,6 +35,71 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
...
+
+	return nil
+}
+
+func (p *Plugin) parseImports() error {
+	for _, pluginsConfigsImports := range p.generateCfg.PluginsConfigsImports {
+		cfg := new([]*SchemaConfig)
+		err := mapstructure.Decode(pluginsConfigsImports[SchemasConfigsKey], cfg)
+
+		if err != nil {
```
Comment (incu6us) [inline generator/plugins/graphql/plugin.go]: inline?

File: generator/plugins/graphql/plugin.go [hunk 8]
```diff
@@ -35,6 +35,71 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
...
+			return errors.Wrap(err, "failed to decode config")
+		}
+
+		for _, schema := range *cfg {
+			err := p.parseImportedSchema(schema)
+
+			if err != nil {
```
Comment (incu6us) [inline generator/plugins/graphql/plugin.go]: inline?

File: generator/plugins/graphql/plugin.go [hunk 9]
```diff
@@ -35,6 +35,71 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
...
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (p *Plugin) parseImportedSchema(cfg *SchemaConfig) error {
+	var importedSchemaName = cfg.Name
```
Comment (incu6us) [inline generator/plugins/graphql/plugin.go]: `cfg` could be nil, so I would like to suggest you to check the state

File: generator/plugins/proto2gql/plugin.go [hunk 10]
```diff
@@ -39,12 +39,35 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
 	}
 	p.generateConfig = config
 	p.config = cfg
+
+	err = p.parseImports()
+	if err != nil {
```
Comment (incu6us) [inline generator/plugins/proto2gql/plugin.go]: inline?

File: generator/plugins/proto2gql/plugin.go [hunk 11]
```diff
@@ -39,12 +39,35 @@ func (p *Plugin) Init(config *generator.GenerateConfig, plugins []generator.Plug
...
+
+	return nil
+}
+
+func (p *Plugin) parseImports() error {
+	for _, pluginsConfigsImports := range p.generateConfig.PluginsConfigsImports {
+		cfg := new([]*ProtoFileConfig)
+		err := mapstructure.Decode(pluginsConfigsImports["proto2gql_files"], cfg)
+
+		if err != nil {
```
Comment (incu6us) [inline generator/plugins/proto2gql/plugin.go]: inline?

File: generator/plugins/proto2gql/plugin.go [hunk 12]
```diff
@@ -63,20 +86,26 @@ func (p *Plugin) normalizeGenerateConfigPaths() error {
 		p.config.Files[i].ProtoPath = normalizedPath
 
 	}
+
 	return nil
 }
+
 func (p *Plugin) prepareFileConfig(fileCfg *ProtoFileConfig) error {
```
Comment (incu6us) [inline generator/plugins/proto2gql/plugin.go]: do you really need the error on the returning?

File: cmd/go2gql/main.go (hunk 3 above)
Comment (saturn4er) [reply cmd/go2gql/main.go]:
```suggestion
			if err := gc.ParseImports(); err != nil {
			     return errors.Wrap(err, "failed to parse imports")
			}
```

File: cmd/go2gql/main.go (hunk 1 above)
Comment (damour) [reply cmd/go2gql/main.go]: Changed to log.Fatal() after app.Run().

File: generator/config.go (hunk 2 above)
Comment (damour) [reply generator/config.go]: done

File: generator/config.go (hunk 4 above)
Comment (damour) [reply generator/config.go]: done, thanks

File: generator/plugins/proto2gql/plugin.go (hunk 6 above)
Comment (damour) [reply generator/plugins/proto2gql/plugin.go]: done

File: cmd/go2gql/main.go (hunk 3 above)
Comment (damour) [reply cmd/go2gql/main.go]: done

File: generator/plugins/graphql/plugin.go (hunk 5 above)
Comment (damour) [reply generator/plugins/graphql/plugin.go]: done

File: generator/plugins/graphql/plugin.go (hunk 7 above)
Comment (damour) [reply generator/plugins/graphql/plugin.go]: done

File: generator/plugins/graphql/plugin.go (hunk 8 above)
Comment (damour) [reply generator/plugins/graphql/plugin.go]: done

File: generator/plugins/graphql/plugin.go (hunk 9 above)
Comment (damour) [reply generator/plugins/graphql/plugin.go]: done, thanks.

File: generator/plugins/proto2gql/plugin.go (hunk 10 above)
Comment (damour) [reply generator/plugins/proto2gql/plugin.go]: +

File: generator/plugins/proto2gql/plugin.go (hunk 11 above)
Comment (damour) [reply generator/plugins/proto2gql/plugin.go]: +

File: generator/plugins/proto2gql/plugin.go (hunk 12 above)
Comment (damour) [reply generator/plugins/proto2gql/plugin.go]: removed.

**Pull request merged**